design/ceph/object/store.md

```diff
@@ -119,3 +119,31 @@ Specifying this section also ensures that the pool section in the ceph-object-zo
   zone:
     name: "name"
 ```
+
+### LDAP
+
+The Ceph Object Gateway supports integrating with LDAP for authenticating and creating users, please refer [here](https://docs.ceph.com/docs/master/radosgw/ldap-auth/). This means that the `ceph object store user` is also required to be part of groups in the LDAP server otherwise, authentication will fail. The following settings need to be configured in the RGW server:
+```
+rgw ldap binddn =
+rgw_ldap_secret = /etc/ceph/ldap/bindpass.secret
+rgw ldap uri =
+rgw ldap searchdn =
+rgw ldap dnattr =
+rgw_s3_auth_use_ldap = true
+```
+So the CRD for the Ceph Object Store will be modified to include the above changes:
+```yaml
+ldap:
```

Comment:
Are there any optional settings that the user may want to apply? If there are very many more I'd suggest we implement this as a `map[string]string` instead of defining them as individual string settings in the types.go, to allow for full flexibility of the ldap settings. But if this is the whole list, we don't need that flexibility.

Reply:
IMO the LDAP settings are minimal and straightforward, chances are less to have more options in future with RGW.
@jhoblitt @BlaineEXE any thoughts??

Reply:
I believe `rgw_ldap_searchfilter` is the only practically optional parameter.  `rgw_ldap_secret` might technically be optional if it is possible to bind to the ldap server without auth.